File: common/src/main/java/work/fking/pangya/common/model/server/ServerFlag.java
package work.fking.pangya.common.model.server;

public enum ServerFlag {
    HIDDEN(1 << 4),
    SORT_PRIORITY(1 << 7),
    GRAND_PRIX(1 << 11);

    private final int value;

    ServerFlag(int value) {
        this.value = value;
    }

    public int value() {
        return value;
    }
}

File: common/src/main/kotlin/work/fking/pangya/common/server/ServerFlag.kt
package work.fking.pangya.common.server

enum class ServerFlag(private val value: Int) {
    HIDDEN(1 shl 4),
    SORT_PRIORITY(1 shl 7),
    GRAND_PRIX(1 shl 11);

    fun value(): Int {
        return value
    }
}
